### PR TITLE
ability to pass jdbc-opts to connect and in-transaction

### PR DESCRIPTION
**Pull request by jehaby**
this PR allows passing jdbc-options map to `connect` & `in-transaction` functions. 

this functionality needed for this PR to frankie
https://github.com/Flexiana/Frankie/pull/488

### CODE REVIEW

File: src/framework/db/core.clj [hunk 1]
```diff
@@ -61,8 +66,13 @@
...
   ([config count]
-   (try (let [db (:framework.db.storage/postgresql config)
-              mig-config (assoc (:framework.db.storage/migration config) :db db)]
+   (try (let [db-conf {:datasource (-> config
+                                       :framework.db.storage/postgresql
+                                       :datasource
+                                       jdbc/get-datasource)}
+              mig-config (assoc (:framework.db.storage/migration config)
+                                :db db-conf)]
+          (def mc mig-config)
```
Comment (g-krisztian) [inline src/framework/db/core.clj:75]: Could you delete this line?

File: src/framework/db/core.clj [hunk 2]
```diff
@@ -91,23 +101,23 @@
...
+    (jdbc/execute! datasource sql-params default-opts)))
 
 (defn in-transaction
-  [tx sql-map]
-  {:pre [(instance? PgConnection tx)]}
-  (let [sql-params (->sql-params sql-map)]
-    (jdbc/execute! tx sql-params {:return-keys true})))
+  ([tx sql-map]
+   (in-transaction tx sql-map nil))
+  ([tx sql-map jdbc-opts]
```
Comment (g-krisztian) [inline src/framework/db/core.clj:110]: Frankie has some direct call to this function, those are updated too?

Comment (g-krisztian) [reply src/framework/db/core.clj:110]: I see it now

File: src/framework/db/core.clj (hunk 1 above)
Comment (jehaby) [reply src/framework/db/core.clj]: thx, done!

**Review by g-krisztian — APPROVED** [review]

**Review by bigodel — APPROVED** [review]

**Pull request merged**